script.sh

```diff
@@ -97,8 +97,18 @@ echo '::group:: Running tflint with reviewdog 🐶 ...'
   # Allow failures now, as reviewdog handles them
   set +Eeuo pipefail
 
+
+  # We only want to specify the tflint target directory if it is not the default to avoid conflicts
+  CHDIR_COMMAND=""
+  if [ "$INPUT_TFLINT_TARGET_DIR" == "." ]; then
+    echo "Using default working directory. No need to specify chdir"
+  else
+    echo "Custom target directory specified."
+    CHDIR_COMMAND="--chdir=${INPUT_TFLINT_TARGET_DIR}"
+  fi
+
   # shellcheck disable=SC2086
-  TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR} "${TFLINT_PATH}/tflint" -c "${INPUT_TFLINT_CONFIG}" --format=checkstyle ${INPUT_FLAGS} --chdir=${INPUT_TFLINT_TARGET_DIR} \
+  TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR} "${TFLINT_PATH}/tflint" -c "${INPUT_TFLINT_CONFIG}" --format=checkstyle "${INPUT_FLAGS}" "${CHDIR_COMMAND}" \
```

Comment:
**issue:** We shouldn't surround `INPUT_FLAGS` and `CHDIR_COMMAND` with quotes in this particular case.

To show why, I can give an example. If `CHDIR_COMMAND` is set to an empty string (`""`), then the line
```bash
tflint ${CHDIR_COMMAND}
```
evaluates to:
```bash
tflint
```
which is a valid `tflint` command. OTOH, the line:
```bash
tflint "${CHDIR_COMMAND}"
```
would evaluate to
```bash
tflint ""
```
which would cause `tflint` to fail with the following error message:
```
Failed to parse CLI arguments; Failed to load ``: File not found
```
and a nonzero exit code.

ShellCheck normally complains if shellscripts contain variables that are not quoted, which is why we have added the `shellcheck disable=SC2086` line here.

Reply:
That makes sense - thanks